### ISSUE TITLE
Copy/fork the spambayes Classifier class (and dependencies)

### ISSUE THREAD

**Issue by BjarniRunar** [mailpile/Mailpile#1991]
In order to simplify future porting to Python 3.x (Issue #160) and to avoid an external dependency on an unmaintained library, we should probably just fork the `spambayes.Classifier` class and copy it (and its dependencies) into our tree.

I propose we call it `mailpile.spambayes.Classifier` - to give credit where credit is due, if nothing else.

The port/fork should make a point of deleting as much code as possible, but leave copyrights and credits intact. There are all sorts of things in there, including URL retrieval (!) and a tokenizer, which we are not using. Less code means less maintenance, fewer places for bugs to hide and less work porting when the time comes.

**Comment by deepaknalore** [comment]
Can I take this up? 
@BjarniRunar Can you please give me more insights on how to go about this?

**Comment by BjarniRunar** [comment]
I'm very sorry about the late response; If you @deepaknalore, or anyone else are still interested; I don't have any clearer ideas about how exactly to do this, than those stated in the initial issue.

Basically, Mailpile is only using a tiny fraction of spambayes. This issue basically requires just copying as little as possible from the spambayes source into Mailpile's own tree and then porting it so it a) works and b) is compatible with Python 3.x.

Steps a) and b) don't have to happen at the same time. And probably shouldn't.

**Comment by vezult** [comment]
Just to see what this would entail, I stripped everything out of spambayes except for the classifier module and its direct dependencies:

https://github.com/vezult/spambayes/tree/strip_for_mailpile

I also tried to make it work with python3, avoiding any invasive changes. I'm sure this means that my approach in some cases leave something to be desired, but it seems to have been effective. The tests seem to run successfully in Python 3.5.3, and Python 2.7.13, in my testing (`python Tester.py`).

If I attempt to import this work into mailpile, would you rather I import the original files, and then apply my patches to get them working, or import the files in their current "working" state, and do any additional changes from there?

**Comment by BjarniRunar** [comment]
Hey @vezult that's awesome! Thank you!

I think my preference would be to import them in their original state and then commit your changes, just in case someone wants to go exploring, and then add integration commits on top of that.

Note that committing this to the Mailpile tree directly is the lazy short-cut; the more community friendly way would be to create a new package in PyPI so other projects could make use of it too. Bayesian classifiers are widely applicable and useful things! But I am 100% OK with taking the short-cut, if someone wants to factor it out into a project of its own later that's not exactly hard.

**Comment by vezult** [comment]
@BjarniRunar yeah, agreed. I currently lack sufficient familiarity with the code to be confident about publishing the code as is. I'd be happier if, after integration with mailpile, it does indeed work as intended :-). I ran the tests, and they seemed to pass, but I don't really know how much coverage that represents, or what the really means for real usage.

Anyway, I've moved my work into a [mailpile branch](https://github.com/vezult/Mailpile/tree/issue1991_partial_import_of_spambayes), and have run `make alltests`, and everything seems to pass for me except for the crypto/streamer tests, which I imagine should be unrelated.

It looks as if the classifier is used by the auto tagging feature. I'll be honest...I've not used mailpile for a while. Does anyone have some guidance/suggestions/pointer to docs on how to test that it is working? That might give me a leg up. Otherwise, I'll try to work it out.
